libarchive_fe/passphrase.c

```diff
@@ -162,7 +162,11 @@ handler(int s)
 	signo[s] = 1;
 }
 
+#ifdef __HAIKU__
+char *
```

Comment:
Why can't this be static on Haiku?  What is the error?

Reply:
I get this error after "./configure && make" (Haiku beta5, gcc 13.3.0):

```
  CC       libarchive_fe/la-passphrase.lo
libarchive_fe/passphrase.c:166:1: error: static declaration of 'readpassphrase' follows non-static declaration
  166 | readpassphrase(const char *prompt, char *buf, size_t bufsiz, int flags)
      | ^~~~~~~~~~~~~~
In file included from libarchive_fe/passphrase.c:61:
/boot/system/develop/headers/bsd/readpassphrase.h:41:8: note: previous declaration of 'readpassphrase' with type 'char *(const char *, char *, size_t,  int)' {aka 'char *(const char *, char *, long unsigned int,  int)'}
   41 | char * readpassphrase(const char *, char *, size_t, int);
      |        ^~~~~~~~~~~~~~
Makefile:7117: recipe for target 'libarchive_fe/la-passphrase.lo' failed
make[1]: *** [libarchive_fe/la-passphrase.lo] Error 1
make[1]: Leaving directory '/boot/home/develop/libarchive-3.7.5'
Makefile:4084: recipe for target 'all' failed
make: *** [all] Error 2
```

Reply:
Ah.  I see.  This is the wrong fix for this.  The problem is that the Haiku system headers define a system version of `readpassphrase` and we're not detecting it.  If we detected it, then `HAVE_READPASSPHRASE` would be set and we would not be trying to define our own version of this function.

Reply:
By the way, I'm happy to merge everything else here as-is.  It's up to you whether  you want to wait until you can work out the proper fix for this PR or back out this one change and submit it later.

Reply:
Ah - Haiku has the header file but fails on linking, according to `config.log`.
Thus, calling `./configure LIBS="-lbsd"` fixes the problem on Haiku.
The question is whether it makes sense to detect this on Haiku and add `-lbsd` automatically
(I guess this would be some kind of separate FR)

So it's ok for me to back out the change for Haiku.

Reply:
Ok, so this change:

```
--- configure.ac.orig   2024-09-24 12:22:57.492830720 +0000
+++ configure.ac        2024-09-24 12:24:44.687079424 +0000
@@ -97,6 +97,7 @@
 case "$host_os" in
   *mingw* ) inc_windows_files=yes ;;
   *cygwin* | *msys*) inc_cygwin_files=yes ;;
+  haiku*) LIBS="-lbsd $LIBS" ;;
 esac
 AM_CONDITIONAL([INC_WINDOWS_FILES], [test $inc_windows_files = yes])
 AM_CONDITIONAL([INC_CYGWIN_FILES], [test $inc_cygwin_files = yes])
```

in `configure.ac` makes it compile on Haiku w/o specifying `LIBS`

In case you confirm that this is a valid solution, I would raise a new PR.

Reply:
Yes, I think that makes sense.  Do we need something similar for the Cmake build?

CC: @evelikov (who wants to add yet another build system)

Reply:
PR #2352  created

Reply:
So this should not be needed now after #2352 was merged?  If not, then please revert this change and everything else here looks good.

Reply:
Correct, the Haiku part can be removed here (I reverted it)